internalsdk/survey.go

```diff
@@ -0,0 +1,104 @@
+package internalsdk
+
+import (
+	"encoding/json"
+	"io"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/getlantern/errors"
+	"github.com/getlantern/flashlight/v7/proxied"
+)
+
+var (
+	dialTimeout = 30 * time.Second
+	httpClient  = &http.Client{
+		Transport: proxied.ParallelPreferChained(),
+		Timeout:   dialTimeout,
+	}
+
+	surveyUrl     = "https://raw.githubusercontent.com/getlantern/loconf/master/messages.json"
+	noSurveyError = errors.New("Survey is not available for the user")
+)
+
+type SurveyResponse struct {
+	DefaultLocale *string           `json:"defaultLocale,omitempty"`
+	Surveys       map[string]Survey `json:"surveys,omitempty"`
+}
+
+type Survey struct {
+	Enabled         bool    `json:"enabled"`
+	Probability     float64 `json:"probability"`
+	Campaign        string  `json:"campaign"`
+	URL             string  `json:"url"`
+	Message         string  `json:"message"`
+	Thanks          string  `json:"thanks"`
+	Button          string  `json:"button"`
+	ShowPlansScreen *bool   `json:"showPlansScreen,omitempty"`
+}
+
+type SurveyModel struct {
+	*SessionModel
+}
+
+var surveysResponse *SurveyResponse
```

Comment:
Is it possible to update fetchSurvey to return a SurveyResponse instead of using a global variable here?

If it's to cache the survey response, I don't think that's necessary since we only fetch the survey once right now after Lantern starts anyway.

Reply:
fetchSurvey fetches surveys only once and IsSurveyAvalible method checks if surveys are available to the user or not, so I need to store the survey, so all logic stays in one method and we can reuse that across all platforms.